internal/helm/controller/controller.go

```diff
@@ -80,7 +83,20 @@ func Add(mgr manager.Manager, options WatchOptions) error {
 
 	o := &unstructured.Unstructured{}
 	o.SetGroupVersionKind(options.GVK)
-	if err := c.Watch(&source.Kind{Type: o}, &libhandler.InstrumentedEnqueueRequestForObject{}); err != nil {
+
+	var preds []ctrlpredicate.Predicate
+
+	// If a selector has been specified in watches.yaml, add it to the watch's predicates.
+	if !reflect.ValueOf(options.Selector).IsZero() {
```

Comment:
Last thing: it would be nice to have at least one simple test where `WatchOptions.Selector` is set either from a `watches.yaml` (ideally), or to some value parsed from a raw string (simulating a `watches.yaml` parse) if that's easier, and `Add()` executes successfully.